### PR TITLE
Updated profileCardResponse handling

### PR DESCRIPTION
**Pull request by imayhaveborkedit**
Fixes copy pasted code from onProfileResponse

### CODE REVIEW

File: handlers/community.js [hunk 1]
```diff
@@ -165,15 +165,9 @@ var onProfileCardResponse = function onProfileCardResponse(message, callback) {
...
-  if (profileCardResponse.result === 1) {
-    if (this.debug) util.log("Received profile data for: " + profileCardResponse.game_account_client.account_id);
-    this.emit("profileCardData", profileCardResponse.game_account_client.account_id, profileCardResponse);
-    if (callback) callback(null, profileCardResponse);
-  }
-  else {
-    if (this.debug) util.log("Received a bad profileResponse");
-    if (callback) callback(profileCardResponse.result, profileCardResponse);
-  }
+  if (this.debug) util.log("Received profile card data for: " + profileCardResponse.game_account_client.account_id);
```
Comment (howardchung) [inline handlers/community.js]: I'm actually not even sure game_account_client exists in the response anymore.  Wouldn't this crash?

Has anyone actually tried running this with debug on?

**Comment by imayhaveborkedit** [comment]
Sorry, missed that.

File: handlers/community.js [hunk 2]
```diff
@@ -165,15 +165,9 @@ var onProfileCardResponse = function onProfileCardResponse(message, callback) {
...
+  this.emit("profileCardData", profileCardResponse.game_account_client.account_id, profileCardResponse);
+  if (callback) callback(null, profileCardResponse);
```
Comment (howardchung) [inline handlers/community.js]: Probably a debug string to edit as well in the request portion of the code (above the response handlers)

**Comment by howardchung** [comment]
Did you confirm debug mode runs without crashing?  It looks like the old code had game_account_client as well so it would have crashed before if that was an issue.

File: handlers/community.js [hunk 3]
```diff
@@ -165,15 +165,9 @@ var onProfileCardResponse = function onProfileCardResponse(message, callback) {
...
-    this.emit("profileCardData", profileCardResponse.game_account_client.account_id, profileCardResponse);
-    if (callback) callback(null, profileCardResponse);
-  }
-  else {
-    if (this.debug) util.log("Received a bad profileResponse");
-    if (callback) callback(profileCardResponse.result, profileCardResponse);
-  }
+  if (this.debug) util.log("Received profile card data for: " + profileCardResponse.account_id);
+  this.emit("profileCardData", profileCardResponse.account_id, profileCardResponse);
+  if (callback) callback(null, profileCardResponse);
```
Comment (howardchung) [inline handlers/community.js]: Also edit the string in the request section L70

**Comment by imayhaveborkedit** [comment]
Works fine for me now.

**Comment by howardchung** [comment]
signoff (but note that I haven't tried running it.  Does anyone know if there's a more efficient way of doing this besides cloning, making a new branch, setting a new remote of the pull request repo's url, and pulling from there?)

@Crazy-Duck, @paralin and maybe @jimmydorry should review

**Comment by Crazy-Duck** [comment]
I'll try and do some tests today or later this weekend to see if this works. As for another way to test pull requests, thats about the only way I see as well :/

**Comment by jimmydorry** [comment]
I haven't tried this method yet, but this was the first google result. Will confirm this weekend (if i don't forget).

https://gist.github.com/piscisaureus/3342247

**Comment by jimmydorry** [comment]
I haven't tried this method yet, but this was the first google result for locally checking out a remote PR. Will confirm this weekend (if i don't forget).

https://gist.github.com/piscisaureus/3342247

**Comment by Crazy-Duck** [comment]
ACK

I checked out the pull request, it seems to be working as it should now. When you send a bogus account_id it sends back a `CMsgDOTAProfileCard` with only the `account_id` filled out. I am going to assume that this is the intended behaviour of the function.

**Comment by jimmydorry** [comment]
Good enough for me.

**Pull request merged**